File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarraysGreaterThanK(vector<int>& a, int k)
    {
        int count=0;
        int i=0,odd=0;
        for(int j=0;j<a.size();j++)
        {
            if(a[j]%2==1)
            odd++;

            while(odd>=k)
            {
                count+=a.size()-j;
                if(a[i]%2==1)
                odd--;
                i++;
            }
        }
        return count;
    }
    
    int numberOfSubarrays(vector<int>& a, int k) 
    {
        return numberOfSubarraysGreaterThanK(a,k)-numberOfSubarraysGreaterThanK(a,k+1);
    }
};

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.java
class Solution {
    private int atLeastKNiceSubarrays(int[] nums, int k) {
        int count = 0, oddCount = 0, i = 0;
        for(int j = 0; j < nums.length; j++){
            oddCount += nums[j] % 2 == 0 ? 0 : 1;
            while(oddCount >= k){
                count += nums.length - j;
                oddCount -= nums[i] % 2 == 0 ? 0 : 1;
                i++;
            }
        }
        return count;
    }
    public int numberOfSubarrays(int[] nums, int k) {
        return atLeastKNiceSubarrays(nums, k) - atLeastKNiceSubarrays(nums, k + 1);
    }
}